pkg/compact/compact.go

```diff
@@ -923,24 +966,20 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 		if err != nil {
 			return errors.Wrap(err, "build compaction groups")
 		}
-		finishedAllGroups := true
-		for _, g := range groups {
-			shouldRerunGroup, _, err := g.Compact(ctx, c.compactDir, c.comp)
-			if err == nil {
-				if shouldRerunGroup {
-					finishedAllGroups = false
-				}
-				continue
-			}
 
-			if IsIssue347Error(err) {
-				if err := RepairIssue347(ctx, c.logger, c.bkt, err); err == nil {
-					finishedAllGroups = false
-					continue
-				}
+		// Send all groups found during this pass to the compaction workers.
+		for _, g := range groups {
```

Comment:
Do we actually need this loop?

What I mean is, maybe it's possible to do:
```
for _, g := range groups{
    errGroup.Go(func(group ) error {
	shouldRerunGroup, _, err := g.Compact(errGroupCtx, c.compactDir, c.comp)
	if err == nil {
		if shouldRerunGroup {
			mtx.Lock()
		        finishedAllGroups = false
			mtx.Unlock()
		}
		continue
	}
...
})(g)
}
```

Reply:
I'm not sure I understand?  This would create a goroutine for every group, but we only want to create `c.concurrency` many goroutines.  

Is there a way to specify that the `errGroup` should run at most `c.concurrency` workers at the same time?  That would be nice, but I didn't think it was built in, which is why I instead explicitly set up the workers on L917 and then explicitly send work to them on L969.

Reply:
Ignore me, you are right :+1: